src/backend/catalog/aclchk.c

```diff
@@ -3006,8 +3007,8 @@ pg_database_aclmask(Oid db_oid, Oid roleid,
 	Oid			ownerId;
 	cqContext  *pcqCtx;
 
-	/* Superusers or on QE bypass all permission checking. */
-	if (GP_ROLE_EXECUTE == Gp_role || superuser_arg(roleid))
```

Comment:
By the way, I know many people like to write "GP_ROLE_EXECUTE == Gp_role" similar code to avoid typo like "GP_role = GP_ROLE_EXECUTE" but gcc will warn against this kind of error.